src/real/analysis/real_topologyScript.sml

```diff
@@ -1428,7 +1412,7 @@ val EXCHANGE_LEMMA = store_thm ("EXCHANGE_LEMMA",
       FINITE t /\ independent s /\ s SUBSET span t ==>
       ?t'. t' HAS_SIZE CARD t /\ s SUBSET t' /\ t' SUBSET s UNION t /\
         s SUBSET span t') ==>
-    (!t'':real->bool s':real->bool'. (CARD (t'' DIFF s') < CARD (t DIFF s)) ==>
+    (!t'':real->bool s':real->bool. (CARD (t'' DIFF s') < CARD (t DIFF s)) ==>
```

Comment:
This PR contains a strange fix that I don't know how to explain. Here at line 1431 (old) there was an extra apostrophe after the type `bool` but for long time HOL parser didn't report any error.   Now, perhaps it's because I removed a few `open`s at the beginning, HOL starts to report parsing error on this line, then I had to remove this extra char.    Do you have any explaination? @mn200

Reply:
The code with the extra apostrophe actually continues to work on my most recent checkout of HOL.  What's going on is apparent if you do 
```
> “!b:bool'. b /\ p”;
<<HOL message: inventing new type variable names: 'a>>
val it = “∀b '. b ∧ p”: term
```

We see that the apostrophe is interpreted as an extra variable at the level of the quantifiers, not as part of the `bool` token.  Of course, it's vacuous, so in many contexts you may never realise that this is happening.  I'll change the way that types are lexed to fix this.